src/library-authoring/library-sidebar/LibrarySidebar.jsx

```diff
@@ -0,0 +1,54 @@
+// @ts-check
+import React, { useContext } from 'react';
+import {
+  Stack,
+  Icon,
+  IconButton,
+} from '@openedx/paragon';
+import { Close } from '@openedx/paragon/icons';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import messages from '../messages';
+import { AddContentContainer } from '../add-content';
+import { LibraryContext } from '../common/context';
+
+/**
+ * Sidebar container for library pages.
+ *
+ * It's designed to "squash" the page when open.
+ * Uses `sidebarBodyComponent` of the `store` to
+ * choose which component is rendered.
+ * You can add more components in `bodyComponentMap`.
+ * Use the slice actions to open and close this sidebar.
+ */
+const LibrarySidebar = () => {
+  const intl = useIntl();
+  const { sidebarBodyComponent, closeLibrarySidebar } = useContext(LibraryContext);
+
+  const bodyComponentMap = {
+    'add-content': <AddContentContainer />,
+  };
+
+  const buildBody = () => bodyComponentMap[sidebarBodyComponent];
+
+  return (
+    <div className="p-2 vh-100">
+      <Stack direction="horizontal" className="d-flex justify-content-between">
+        <span className="font-weight-bold m-1.5">
+          {intl.formatMessage(messages.addContentTitle)}
+        </span>
+        <IconButton
+          src={Close}
+          iconAs={Icon}
+          alt={intl.formatMessage(messages.closeButtonAlt)}
+          onClick={closeLibrarySidebar}
+          variant="black"
+        />
+      </Stack>
+      {buildBody()}
+    </div>
+  );
+};
+
+LibrarySidebar.propTypes = {};
```

Comment:
Ditto. We also need to migrate this to `tsx`
```suggestion
```

Reply:
Updated https://github.com/open-craft/frontend-app-course-authoring/pull/38/commits/d37ffbf4000b14cfe5ea3065056e7c3d1da1f4e5